### DISCUSSION TITLE
Working on ELS 4.0

### DISCUSSION THREAD

**Discussion by Corionis** [GrokSoft/ELS#41] [General 1. Discussions]
Have been working on ELS 4.0 and the new Navigator desktop application since September 2021. It's a major undertaking -- in the Version-4.0.0 branch. It is not done, packaged or thoroughly tested. Bugs are expected. See the branch [README.md](https://github.com/GrokSoft/ELS/tree/Version-4.0.0) and artifacts/document/ files [developer.md](https://github.com/GrokSoft/ELS/blob/Version-4.0.0/artifacts/document/developer.md) and [release-notes.md](https://github.com/GrokSoft/ELS/blob/Version-4.0.0/artifacts/document/release-notes.md) for more information.

The Navigator Browser tab works locally and remotely with an ELS subscriber listener. The Backup and Libraries tabs are still empty. The Junk Remover tool is done and Jobs work. The Renamer tool is in-progress.

Currently side-tracked implementing a beta version of the 4.0 command line Backup tool for my family with fully bi-directional Hints and the Hint Status Server on a schedule. It is ELS implemented as it was designed taking full advantage of all the _current_ capabilities. Using my family as guinea pigs for version 4.0 :stuck_out_tongue_winking_eye: Eventually the scripts I'm using with be replaced with Jobs.

Along the way I've encountered some rough spots related to background scheduled automation of more complex sequences. Currently smoothing those out, then will return to the Renamer tool and on from there. There's quite a bit of work yet to be done, but it's fairly useful now.

_It's all built-in._

![els-screenshot](https://user-images.githubusercontent.com/21012251/185531513-3574d8cb-5f0e-4f64-bd5b-5b91a40471db.png)

**Comment by GrokSoft** [comment]
Very cool -thanks
